### ISSUE TITLE
Training strategy: why train all layers got lower loss than train only heads.

### ISSUE THREAD

**Issue by nattaon** [matterport/Mask_RCNN#2610]
I've read that there is a training strategy that separate training into 3 stages, 
1. Train heads for few epoch until overfitting, then
2. Train layer 5+ or 4+ for few epoch until overfitting, then
3. Train all layers until overfitting.

But I don't understand, when do you decide overfitting.
Is it the epoch that validation loss increase?

Since I'm not sure when to decide overfitting, I've train each training parameter (all, s3+, s4+, s5+, heads) to see how they are difference.
![loss_all_train100ep](https://user-images.githubusercontent.com/11454194/122887058-7908f380-d37b-11eb-9f84-0e252abb6f7e.png)
My dataset contains  400 training image and 80 validation image of 6 classes. I use the default config parameter with mask_rcnn_coco.h5 pre-trained weight, no learning rate reducing.
In the image 1_train_all got the lowest validation loss, is this mean that for my dataset training all layer is better than train heads first?
Or the training loss and validation loss is not matter, it depended on the prediction result?

**Comment by elivety** [comment]
I have read somewhere that your training folder must have tree fourth to one fourth in the validation folder. 
For your Example 6 classes:   (400/6=) 66 Pictures per class1 in training folder  and (66/4=) 17 pictures of class1 in the validation folder, which you have annotated. 
I trained only the heads, and had good training results on ms coco displayed over tensorboard, no overfitting.
